### PR TITLE
allow messages to be send on ctrl+enter

### PR DESCRIPTION
**Pull request by florianorben**
This adds an option to send chat messages on `ctrl+enter` (`enter` then inserts a new line).
Default option is "as-is", i.e. `enter` sends the message.

### CODE REVIEW

**Comment by it33** [comment]
Thanks @florianorben!

While this functionality looks useful, I'm not sure if "General Settings" is the ideal place for the option...

Would you be open to creating an "Advanced Options" tab at the bottom of Account Settings for this? 

The "Appearance" tab could be a good template, and instead of a radio button perhaps use a checkbox to add this item?

File: web/react/components/user_settings/user_settings_general.jsx [hunk 1]
```diff
@@ -570,6 +585,76 @@ export default class UserSettingsGeneralTab extends React.Component {
...
+                                <input
+                                    type='radio'
+                                    onChange={() => this.updateCtrlSend('false')}
+                                    checked={ctrlSendActive[1]}
+                                />
+                                {'Off'}
+                            </label>
+                            <br/>
+                        </div>
+                        <div><br/>{'If enabled \'Enter\' inserts a new line and ctrl + enter submits the message.'}</div>
```
Comment (it33) [inline web/react/components/user_settings/user_settings_general.jsx]: Proposed edit: `'If enabled \'Enter\' inserts a new line and \'Ctrl + Enter\' submits the message.'`

**Comment by florianorben** [comment]
Hey @it33!

I initially wanted to avoid adding a new tab just for one setting - but I guess lots of settings will be added anyway, so yeah "advanced" sounds good - I'll add that one!

I'm not sure about using a checkbox though, I pretty much used Settings->Notifications as a template (which would also be a better fit for advanced settings I guess?) - there already are some boolean options "Desktop sound notifications" and "Email notifications", which use radio buttons to toggle state.
Using a checkbox could probably break the currently used UX? What do you think?

Agree with your proposed edit for the description :)

**Comment by it33** [comment]
Thanks @florianorben, 

Yes, agree, `Settings -> Notifications` makes sense too. I was thinking if there are a lot of advanced options, Appearance could offer a more compact view--but `Settings -> Notifications` is more standard and flexible, and it's the better design to keep the Advanced Options section from growing too big, 

Good idea :)

**Comment by hmhealey** [comment]
This looks like an awesome feature, but it sounds like there's more work to be done on it to move it to a new tab.

**Comment by florianorben** [comment]
All done + move setting from user.props-store to preference-store

**Comment by it33** [comment]
Thanks @florianorben!

**Comment by crspeller** [comment]
+1

**Comment by jwilander** [comment]
+1

**Pull request merged**